src/ApplicationLogs/LogReader.cs

```diff
@@ -50,173 +67,391 @@ protected override void Configure()
 
         protected override void OnSystemLoaded(NeoSystem system)
         {
-            if (system.Settings.Network != Settings.Default.Network) return;
+            if (system.Settings.Network != Settings.Default.Network)
+                return;
             string path = string.Format(Settings.Default.Path, Settings.Default.Network.ToString("X8"));
-            _db = system.LoadStore(GetFullPath(path));
+            var store = system.LoadStore(GetFullPath(path));
+            _neostore = new NeoStore(store);
+            _neosystem = system;
             RpcServerPlugin.RegisterMethods(this, Settings.Default.Network);
+
+            if (Settings.Default.Debug)
+                ApplicationEngine.Log += OnApplicationEngineLog;
         }
 
+        #endregion
+
+        #region JSON RPC Methods
+
         [RpcMethod]
         public JToken GetApplicationLog(JArray _params)
         {
-            UInt256 hash = UInt256.Parse(_params[0].AsString());
-            byte[] value = _db.TryGet(hash.ToArray());
-            if (value is null)
-                throw new RpcException(-100, "Unknown transaction/blockhash");
+            if (_params == null || _params.Count == 0)
+                throw new RpcException(-32602, "Invalid params");
+            if (UInt256.TryParse(_params[0].AsString(), out var hash))
```

Comment:
Allow to call it with index?

Reply:
That'd be an extension (yes, I will complain about _every_ extension, especially if it's mixed with other code and doesn't have a specific PR for it, it's painfully hard to keep API compatibility when such changes occur randomly, so I'm very happy to see https://github.com/neo-project/neo-modules/pull/865#issuecomment-1885550772 for the current state of PR) and a bit unobvious one, blocks can be indexed, so we usually accept indexes/hashes where blocks are used, but we accept transaction hashes here as well.